### PR TITLE
Lint：Changed rules for project

### PR DESCRIPTION
**Pull request by JihangGuo**
1. 'use strict'  **need check**
	https://eslint.org/docs/rules/strict
2. cli/lib/dataset and  core/src/util/public.ts has more change **need check**
3. core/src/runnder/help.ts 45 
        https://eslint.org/docs/rules/no-param-reassign

ps: This is just a preliminary plan, which needs your inspection and suggestions

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ docs
 dist
 test
 pipcook_venv
+example
```
Comment (Txiaozhe) [inline .eslintignore:6]: example is the part of this project so need lint too, why add to ignore??

**Comment by Txiaozhe** [comment]
There are some conflicts, could you please fix it :)

**Comment by JihangGuo** [comment]
> Good jobs, as @Txiaozhe mentioned before, could you please rebase from master to fix these conflicts?

I have resolved the conflict, but many formatting problems still exist. need you to check the final plan ：）

**Comment by JihangGuo** [comment]
> There are some conflicts, could you please fix it :)

Sorry，i forget to fetch Remote branch. Now it's settled

**Comment by yorkie** [comment]
```sh
##[error]src/utils/logger.ts(6,10): error TS2614: Module '"../runner"' has no exported member 'PipcookRunner'. Did you mean to use 'import PipcookRunner from "../runner"' instead?
##[error]src/runner/helper.ts(9,10): error TS2614: Module '"."' has no exported member 'PipcookRunner'. Did you mean to use 'import PipcookRunner from "."' instead?
##[error]src/utils/public.ts(226,28): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(227,27): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(230,19): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(230,46): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(231,44): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(232,45): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(236,17): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(237,20): error TS2339: Property 'annotation' does not exist on type 'unknown'.
##[error]src/utils/public.ts(238,20): error TS2339: Property 'annotation' does not exist on type 'unknown'.
```

There are some compilers reported from tsc, could you please take a look?

File: packages/cli/lib/dataset.js [hunk 2]
```diff
@@ -21,94 +21,115 @@ const dataset = async (cmdObj) => {
...
             license: 1,
             file_name: image.file_name,
             coco_url: image.coco_url,
             height: image.height,
             width: image.width,
-            id: imageNum
+            id: index + 1
           });
-          imageNum++;
         });
```
Comment (liangzr) [inline packages/cli/lib/dataset.js:69]: I noticed that the strategy of ids for images, categories, and annotations has changed, each iteration of the image directories resets id to 0, please confirm if it works : )

Comment (yorkie) [reply packages/cli/lib/dataset.js:69]: This PR is to change the coding style, may I ask you to keep it just like before?

**Comment by yorkie** [comment]
Closing because of inactivity, thanks for your work @JihangGuo

**Pull request closed without merging**